rq/job.py

```diff
@@ -1308,6 +1326,39 @@ def finished_job_registry(self):
             self.origin, connection=self.connection, job_class=self.__class__, serializer=self.serializer
         )
 
+    def execute_success_callback(self, death_penalty_class: Type[BaseDeathPenalty], result: Any):
+        """Executes success_callback for a job.
+        with timeout .
+
+        Args:
+            death_penalty_class (Type[BaseDeathPenalty]): The penalty class to use for timeout
+            result (Any): The job's result.
+        """
+        if not self.success_callback:
+            return
+
+        logger.debug('Running success callbacks for %s', self.id)
+        self.heartbeat(utcnow(), self.success_callback_timeout)
```

Comment:
Mind moving `heartbeat` call out of this method? I think it's cleaner to have the worker explicitly call `heartbeat()` in the execution flow.